### DISCUSSION TITLE
Avoid <Route render> or not?

### DISCUSSION THREAD

**Discussion by IanVS** [remix-run/react-router#9325] [v5 to v6 Migration]
I'm going through the migration guide, and one of the steps is:

>Use `<Route children>` instead of `<Route render>` and/or `<Route component>` props

So, I did that, even though it meant getting worse type safety of my params in the process. (see https://github.com/remix-run/react-router/issues/8200#issuecomment-1254166954)

But then I came across the next step, dealing with redirects in switches. 

>If you want to redirect client-side, move your `<Redirect>` into a `<Route render>` prop.

But I just went through and got rid of all those!  Was that not necessary?  Why did I have to remove the render props, if I'm going to introduce more of them?  Do they still work in v6 after all?

**Comment by ryanhiebert** [comment]
I have the same question. I'm not able to confirm it yet, but I suspect that we can skip the `render` prop and use `children` instead. There is [no documented `render` prop in v6](https://reactrouter.com/en/main/route/route), but `<Redirect>` seems to become `<Navigate>`.

**Reply by IanVS** [reply]
I think I might have figured it out.  Basically, the idea is not to have any `render` props that rely on the `routeProps` being passed in.  And then, as long as you're not using those, you can convert from `render={() => <Element/>}` to `element={<Element/>}`.  

But, that's just my guess.  I had to give up on my migration because there's no way to block navigation currently (`Prompt` was removed in v6).
